### PR TITLE
Making the uninstall playbook more flexible

### PR DESCRIPTION
**Pull request by brenton**
This handles stage environments as well as the eventual change of aep3_beta to
aep3

### CODE REVIEW

**Comment by brenton** [comment]
@detiber, QE noticed that the uninstall playbook doesn't handle aep3_beta currently.  What do you think about this?  `egrep` comes with the `grep` package.  I tested it with my aep install from stage.

File: playbooks/adhoc/uninstall.yml [hunk 1]
```diff
@@ -125,13 +125,13 @@
...
-    - shell: docker images | grep {{ item }} | awk '{ print $3 }'
+    - shell: docker images | egrep {{ item }} | awk '{ print $3 }'
       changed_when: False
       failed_when: False
       register: images_to_delete
       with_items:
-        - registry.access.redhat.com/openshift3
-        - registry.access.redhat.com/aep3
+        - registry\.access\.*redhat\.com/openshift3
+        - registry\.access\..*redhat\.com/aep3
```
Comment (detiber) [inline playbooks/adhoc/uninstall.yml]: The double dots here on purpose? I assume it is to catch stage images as well?

Comment (brenton) [inline playbooks/adhoc/uninstall.yml]: the double dots were on purpose.  I wanted to escape all the literal dots but then allow for `.*` to match anything in between.  Looks like I didn't do the same for the openshift3 case. :/

**Comment by xiama** [comment]
@brenton should stop all containers before search the 'Exited' container.

**Comment by brenton** [comment]
@xiama, Hmm... my initial thought was that we would only stop the master and node images as is done here: https://github.com/openshift/openshift-ansible/blob/master/playbooks/adhoc/uninstall.yml#L106. Obviously restarting docker like we do at the bottom of that playbook will stop everything else.  Presumably any containers they were running outside of OpenShift would have to be restarted manually.

The playbook is more concerned about guaranteeing that a subsequent install works.  I don't leftover exited containers would cause a problem.  Cleaning up the disk space from the old images and ensuring that a fresh install redownloads the latest images is a lot more important.  That said, if you know of particular edge case we need to handle just let us know!

**Comment by brenton** [comment]
@detiber, are you generally OK with this PR now?

**Comment by detiber** [comment]
@brenton :+1:

**Pull request merged**